### PR TITLE
Fix Helm chart public_addr for proxy when using Ingress

### LINKED ISSUE #4080: connection refused on tsh due to port "mismatch"
### Description

**What happened**:
I've installed teleport via the helm chart and tried to configure it as far as the docs told me what to do. I can login to the web-ui, create new users, enable github, and login to the teleport pod (the only one showing) and get the session recorded.
Once I try to login via tsh, I get this error
```
• tsh login --proxy=teleport.domain.tld --auth=github login
error: Post https://teleport.domain.tld:3080/v1/webapi/github/login/console: dial tcp xxx.xxx.xxx.xxx:3080: connect: connection refused
```
It seems that tsh appends the port, but with an nginx-ingress enabled it just exposes the 443 port, so the connection will never go through.

**What you expected to happen**:
That the tsh login goes through the ingress.

**How to reproduce it (as minimally and precisely as possible)**:

### Environment
- Teleport version (use `teleport version`): Teleport v4.3.0 git:v4.3.0-0-g73b68f401 go1.13.2
- Tsh version (use `tsh version`): Teleport v4.3.0 git:v4.3.0-0-g73b68f401 go1.13.2
- OS (e.g. from `/etc/os-release`): kubernetes, used the helm-chart

- Where are you running Teleport? (e.g. AWS, GCP, Dedicated Hardware):
Kubernetes on root-servers 

**Relevant Debug Logs If Applicable**
 my values.yaml for helm looks like this:
https://gist.github.com/aberfeldy/9444980f09abca6c3b7fd0deffb74cef

Any help would be appreciated

### PR DESCRIPTION
**Pull request by webvictim**
When using an `Ingress` to front your Teleport cluster in Kubernetes, access will be via port 443 (HTTPS) or 80 (HTTP - not recommended). Unfortunately, the Helm chart makes the assumption that you'll always be on port 3080 and this breaks `tsh` logins at the command line. This fixes that problem by specifically overriding the `public_addr` of `proxy_service` when using an `Ingress`.

I also changed all other `public_addr` settings in the main Helm chart to use the configured service port, rather than the `containerPort`. This means that other traffic will be directed correctly if people change the default ports in their config.

I also found a lot of random trailing spaces and blank tabs, so the PR removes all of those that I could find under our Helm charts too.

Fixes #4080

### CODE REVIEW

File: examples/chart/teleport/templates/config.yaml
Comment (webvictim) [inline examples/chart/teleport/templates/config.yaml:74]: Assumptions:
- when TLS is configured on the `Ingress` it'll be using port 443
- when TLS is not configured on the `Ingress` it'll be using port 80

When not using an `Ingress`, we change to using the host-side `proxyweb` port under the service definition rather than the container-side port.

File: examples/chart/teleport/values.yaml
Comment (benarent) [inline examples/chart/teleport/values.yaml:31]: When using Teleport self-signed certs do customers still get a browser warning?  

Now we have this in our docs, it would be good to get a demo / video of this setup.

Comment (webvictim) [reply examples/chart/teleport/values.yaml:31]: > When using Teleport self-signed certs do customers still get a browser warning?

Yes. There isn't really a good way to avoid this when just using Teleport. The simplest way to do it in Kubernetes is to disable TLS termination via Teleport itself, then let an `Ingress` with [`cert-manager`](https://github.com/jetstack/cert-manager) correctly set up automatically provision your certs for you. How to set that up is probably out of scope for Teleport docs, though.

Comment (benarent) [reply examples/chart/teleport/values.yaml:31]: I know it's out of scope for this PR, but I wonder what the best practices are for this. Chrome is really locking down on self-signed certs, and would be good to have a nice story / instructions for this. @awly Do you know of any other OSS project that has his same issue?

Comment (webvictim) [reply examples/chart/teleport/values.yaml:31]: Something that works for _everything_ is pretty hard. Maybe we could provide an easier way to get the self-signed web CA out of Teleport and give instructions on how to import that into the local browser store. It's all just pain though.

It's either that, or we make it easier to run without TLS at all - not great for a security-focused product.

Comment (awly) [reply examples/chart/teleport/values.yaml:31]: We can also consider compiling a [LetsEncrypt client](https://godoc.org/golang.org/x/crypto/acme/autocert) into teleport.
For example, a proxy can attempt to get certs for itself before falling back to self-signed certs.

Examples of this:
- https://caddyserver.com/docs/automatic-https
- https://github.com/haproxytech/haproxy-lua-acme
- https://perkeep.org/doc/server-config
- https://docs.gitea.io/en-us/https-setup/#using-lets-encrypt
- etc https://pkg.go.dev/golang.org/x/crypto/acme/autocert?tab=importedby

**Review by awly — APPROVED** [review]

File: examples/chart/teleport/values.yaml
Comment (benarent) [reply examples/chart/teleport/values.yaml:31]: +1 I like this, but to not block the fix. Let's put it into an issue and work on another PR.

Comment (awly) [reply examples/chart/teleport/values.yaml:31]: https://github.com/gravitational/teleport/issues/4117

Comment (webvictim) [reply examples/chart/teleport/values.yaml:31]: The other alternative for local testing (that @benarent just reminded me of) is something like https://github.com/FiloSottile/mkcert

**Comment by webvictim** [comment]
@benarent Could I have an approval stamp please? I think you're code owner on `*.md`.

**Review by benarent — APPROVED** [review]
Approving this to resolve customers bug , other TLS discussion has been moved to another ticket.

**Pull request merged**
